lib/reverse_string.rb

```diff
@@ -1,5 +1,30 @@
 def reverse_characters(str)
+
+    i = 0    
+    n = str.size - 1
+    ansstr = " "*n
+    
+while(i<str.size)
+    ansstr[i] = str[n-i]
+    i = i + 1
+end
+
+return ansstr
 end
 
 def reverse_words(str)
+    
+    strarr = str.split(" ")
+    ansarrsize = strarr.size
+    i = 0
+    ansarr = Array.new []
```

Comment:
When creating an empty Array, you can use `Array.new` _or_ the shorthand `[]`. No need for both!